src/Transport/Sending/OutgoingMessageExtensions.cs

```diff
@@ -7,16 +7,21 @@
 
     static class OutgoingMessageExtensions
     {
-        public static ServiceBusMessage ToAzureServiceBusMessage(this OutgoingMessage outgoingMessage, DispatchProperties dispatchProperties, string incomingQueuePartitionKey)
+        public static ServiceBusMessage ToAzureServiceBusMessage(this IOutgoingTransportOperation outgoingTransportOperation, DispatchProperties dispatchProperties, string incomingQueuePartitionKey)
         {
+            var outgoingMessage = outgoingTransportOperation.Message;
             var message = new ServiceBusMessage(outgoingMessage.Body)
             {
                 // Cannot re-use MessageId to be compatible with ASB transport that could have native de-dup enabled
                 MessageId = Guid.NewGuid().ToString()
             };
 
-            // The value needs to be "application/octect-stream" and not "application/octet-stream" for interop with ASB transport
-            message.ApplicationProperties[TransportMessageHeaders.TransportEncoding] = "application/octect-stream";
+            var disableLegacyHeaders = outgoingTransportOperation.Properties.ContainsKey(TransportOperationExt.DisableLegacyHeadersKey);
```

Comment:
If I recall correctly, those properties get serialized into the outbox too. It might be worthwhile adding a comment here to indicate why the property is not removed (because it needs to be preserved even with the outbox if I understand this correctly).

From a more general perspective I wonder if it would make sense to actually moved this up to the transport level configuration. For all users that do not need to interop the old transport this header is unnecessary. By moving it to the transport level we can make it opt-out, later opt-in and eventually deprecate it from the transport entirely.

This seems to be the more future-proof way of handling this change instead of introducing yet another configuration layer deep down on the transport operation level.

Reply:
@danielmarbach I like your reasoning about uplifting this and not have it per operation but as an endpoint configuration. Here it would be AND also on the top level configuration as the ***** connector uses the transport seam.

Reply:
Applied on PR